gce/survival-training/README-tf-estimator.md

```diff
@@ -0,0 +1,404 @@
+
+```bash
+python generate_cifar10_tfrecords.py --data-dir=/tmp/cifar-10-data
+```
+
+
+Now make a bucket in which to store this data (note that bucket names have to be globally unique, so we will prefix this with our project name):
+
+```bash
+gsutil mb gs://$(gcloud config get-value project)-cifar-10-data
+```
+
+Now we can upload our data to the bucket:
+
+```bash
+gsutil -m cp -r /tmp/cifar-10-data/*  gs://$(gcloud config get-value project)-cifar-10-data/
+```
+
+You can test that the data has been transferred using
+
+```bash
+gsutil ls gs://$(gcloud config get-value project)-cifar-10-data/
+```
+
+
+### Checkpoints
+
+In the course of training, checkpoints will be stored at the path provided to the trainer using the `--job-dir` argument. We can provide a GCS path for this argument, as well, and we will do so.
+
+Let us make ourselves a checkpoint bucket:
+
+```bash
+gsutil mb gs://$(gcloud config get-value project)-cifar-10-checkpoints
+```
+
+With all this preparation in place, we are ready to specify a startup script and define our instance metadata.
+
+
+## Cloud Source Repositories
+
+The [tensorflow/models](https://github.com/tensorflow/models) repo exposes each of its subdirectories as Python modules, which is really handy. What we are going to do is push a copy of TensorFlow models as one of our own Cloud Source repositories and run the training job from there. This way, if you want to experiment with the model architecture in the future, you can easily do so in your GCE environment (using the same kinds of tagging semantics mentioned in the [README](./README.md)).
+
+To do so, we must create a source repo, which [you can do from Cloud Console here](https://console.cloud.google.com/code/develop/repo). Alternatively, you can use the `gcloud` tool:
+
+```bash
+gcloud source repos create tensorflow-models
+```
+
+If this is the first time you are using source repos, make sure to run the following command from your local machine:
+
+```bash
+git config credential.helper gcloud.sh
+```
+
+Now navigate to your local clone of [tensorflow/models](https://github.com/tensorflow/models) and add this new repo as a remote:
+
+```
+git remote add gcp https://source.developers.google.com/p/$(gcloud config get-value project)/r/tensorflow-models
+```
+
+Finally,
+
+```bash
+git push -u gcp-or-whatever master
+```
+
+This push might take some time because of the size of `tensorflow/models`.
+
+
+## Startup script
+
+We will use the [tf-estimator-startup.sh](./gce/tf-estimator-startup.sh) script, which is only a slight modification of [our origina dummy script](./gce/startup.sh).
```

Comment:
are the modifications meaningful enough to give a short explanation on here?

Reply:
I think they aren't, but am curious how you feel about them. Could you have a look at the two files and give me your opinion?